tools/tesseratos/src/tesseratos/console/plugin.cpp

```diff
@@ -0,0 +1,270 @@
+#include "plugin.hpp"
+
+#include <imgui.h>
+
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/imgui/plugin.hpp>
+
+#include "../toolbox/plugin.hpp"
+
+using cubos::engine::Cubos;
+
+using namespace tesseratos;
+
+namespace
+{
+    // Resource used to store the state of the console
+    struct State
+    {
+        std::vector<cubos::core::Logger::Entry> uiEntries;
+        std::size_t prevCursor = 0; // Previous cursor position~could find a better way to clear log
+        char searchString[256];
+        bool clearEnabled = false;
+        bool collapseEnabled = false;
+        bool traceEnabled = true;
+        bool debugEnabled = true;
+        bool infoEnabled = true;
+        bool warnEnabled = true;
+        bool errorEnabled = true;
+        bool criticalEnabled = true;
+    };
+} // namespace
+
+// Function to convert log level to a string
+static std::string levelToString(cubos::core::Logger::Level level)
+{
+    switch (level)
+    {
+    case cubos::core::Logger::Level::Trace:
+        return "Trace";
+    case cubos::core::Logger::Level::Debug:
+        return "Debug";
+    case cubos::core::Logger::Level::Info:
+        return "Info";
+    case cubos::core::Logger::Level::Warn:
+        return "Warn";
+    case cubos::core::Logger::Level::Error:
+        return "Error";
+    case cubos::core::Logger::Level::Critical:
+        return "Critical";
+    default:
+        return "Unknown";
+    }
+}
+
+// Function to convert log level to a color
+static ImVec4 levelToColor(cubos::core::Logger::Level level)
+{
+    switch (level)
+    {
+    case cubos::core::Logger::Level::Trace:
+        return {1.0F, 1.0F, 1.0F, 1.0F}; // White
+    case cubos::core::Logger::Level::Debug:
+        return {0.0F, 1.0F, 1.0F, 1.0F}; // Cyan
+    case cubos::core::Logger::Level::Info:
+        return {0.0F, 1.0F, 0.0F, 1.0F}; // Green
+    case cubos::core::Logger::Level::Warn:
+        return {1.0F, 1.0F, 0.0F, 1.0F}; // Yellow
+    case cubos::core::Logger::Level::Error:
+        return {1.0F, 0.0F, 0.0F, 1.0F}; // Red
+    case cubos::core::Logger::Level::Critical:
+        return {1.0F, 0.0F, 0.0F, 0.5F}; // Bright Red
```

Comment:
How is this Bright Red? Is it not just the same red as Error, but more transparent?